pstl/include/pstl/internal/algorithm_impl.h

```diff
@@ -1141,26 +1141,28 @@ __brick_count(_ForwardIterator __first, _ForwardIterator __last, _Predicate __pr
     return std::count_if(__first, __last, __pred);
 }
 
-template <class _ExecutionPolicy, class _ForwardIterator, class _Predicate, class _IsVector>
+template <class _Tag, class _ExecutionPolicy, class _ForwardIterator, class _Predicate>
 typename std::iterator_traits<_ForwardIterator>::difference_type
-__pattern_count(_ExecutionPolicy&&, _ForwardIterator __first, _ForwardIterator __last, _Predicate __pred,
-                /* is_parallel */ std::false_type, _IsVector __is_vector) noexcept
+__pattern_count(_Tag, _ExecutionPolicy&&, _ForwardIterator __first, _ForwardIterator __last, _Predicate __pred) noexcept
 {
-    return __internal::__brick_count(__first, __last, __pred, __is_vector);
+    return __internal::__brick_count(__first, __last, __pred, typename _Tag::_IsVector{});
 }
 
-template <class _ExecutionPolicy, class _RandomAccessIterator, class _Predicate, class _IsVector>
+template <class _IsVector, class _ExecutionPolicy, class _RandomAccessIterator, class _Predicate>
 typename std::iterator_traits<_RandomAccessIterator>::difference_type
-__pattern_count(_ExecutionPolicy&& __exec, _RandomAccessIterator __first, _RandomAccessIterator __last,
-                _Predicate __pred,
-                /* is_parallel */ std::true_type, _IsVector __is_vector)
+__pattern_count(__parallel_tag<_IsVector> __tag, _ExecutionPolicy&& __exec, _RandomAccessIterator __first,
+                _RandomAccessIterator __last, _Predicate __pred)
 {
+    using __backend_tag = typename decltype(__tag)::__backend_tag;
+
     typedef typename std::iterator_traits<_RandomAccessIterator>::difference_type _SizeType;
     return __internal::__except_handler([&]() {
         return __par_backend::__parallel_reduce(
-            std::forward<_ExecutionPolicy>(__exec), __first, __last, _SizeType(0),
-            [__pred, __is_vector](_RandomAccessIterator __begin, _RandomAccessIterator __end, _SizeType __value)
-                -> _SizeType { return __value + __internal::__brick_count(__begin, __end, __pred, __is_vector); },
+            __backend_tag{}, std::forward<_ExecutionPolicy>(__exec), __first, __last, _SizeType(0),
+            [__pred, __is_vector](_RandomAccessIterator __begin, _RandomAccessIterator __end,
+                                  _SizeType __value) -> _SizeType {
+                return __value + __internal::__brick_count(__begin, __end, __pred, typename __backend_tag::_IsVector{});
```

Comment:
I believe this is an error.
Should be
```suggestion
                return __value + __internal::__brick_count(__begin, __end, __pred, _IsVector{});
```

Reply:
BTW, In my opinion, `__backend_tag::__is_vector{}` is more right approach with architect perspective.
The type `__backend_tag::__is_vector` is a contract for backend tag type.
Type _IsVector is kind of implementation derails of "__parallel_tag".

Reply:
Functions with some runtime don't work because we cannot change the return type at run-time. We agreed that all tags of the first level should have `__is_vector` and some of them could have `backend_tag` if necessary